File: backend/app/Models/Article/Article.php
<?php

namespace App\Models\Article;

use App\Models\User;
use Eloquent;
use Illuminate\Database\Eloquent\Builder;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

/**
 * @mixin Eloquent
 * @mixin Builder
 */
class Article extends Model
{
  protected $fillable = [
    'text',
    'title',
    'description',
    'preview',
    'author_id',
    'article_category_id',
  ];

  public function author(): BelongsTo
  {
    return $this->belongsTo(User::class, 'author_id');
  }

  public function articleCategory(): BelongsTo
  {
    return $this->belongsTo(ArticleCategory::class);
  }

  public function comments(): HasMany
  {
    return $this->hasMany(Comment::class);
  }
}

File: UserPanel/Models/Article.cs
using System.Text.Json.Serialization;
namespace UserPanel.Models;

public class Article
{
    [JsonPropertyName("id")] public required int Id { get; set; }
    [JsonPropertyName("text")] public required string Text { get; set; }
    [JsonPropertyName("title")] public required string Title { get; set; }
    [JsonPropertyName("description")] public string? Description { get; set; }
    [JsonPropertyName("preview")] public required string Preview { get; set; }
    [JsonPropertyName("article_category")] public ArticleCategory? ArticleCategory { get; set; }
    [JsonPropertyName("comments")] public List<Comment>? Comments { get; set; } = [];
    [JsonPropertyName("author")] public User? Author { get; set; }

}
